src/app/wallets/reimburse-failed-usd.ts

```diff
@@ -0,0 +1,80 @@
+import { toSats } from "@domain/bitcoin"
+import {
+  CouldNotFindWalletForWalletCurrencyError,
+  InvalidAccountForWalletIdError,
+} from "@domain/errors"
+import { DisplayCurrency, toCents } from "@domain/fiat"
+import { LedgerTransactionType } from "@domain/ledger"
+import { AmountCalculator, WalletCurrency } from "@domain/shared"
+
+import { WalletsRepository } from "@services/mongoose"
+import * as LedgerFacade from "@services/ledger/facade"
+
+const calc = AmountCalculator()
+
+export const reimburseFailedUsdPayment = async <
+  S extends WalletCurrency,
+  R extends WalletCurrency,
+>({
+  paymentFlow,
+  journalId,
+  accountId,
+}: {
+  paymentFlow: PaymentFlow<S, R>
+  journalId: LedgerJournalId
+  accountId: AccountId
+}): Promise<true | ApplicationError> => {
+  const paymentHash = paymentFlow.paymentHashForFlow()
+  if (paymentHash instanceof Error) return paymentHash
+
+  const metadata: FailedPaymentLedgerMetadata = {
+    hash: paymentHash,
+    type: LedgerTransactionType.Payment,
+    pending: false,
+    related_journal: journalId,
+
+    usd: Number(
```

Comment:
somewhat a duplicate from `centsAmount`?

Reply:
Yup, but we need this duplicate in for now for consistency with the `LedgerTransaction` type across other non-`"payment"` tx types. Will be addressed as part of the cleanup [here](https://github.com/GaloyMoney/galoy/projects/9#card-84279899).